packages/adblocker/src/engine/bucket/preprocessor.ts

```diff
@@ -0,0 +1,161 @@
+import { StaticDataView } from '../../data-view';
+import Preprocessor, { Env } from '../../preprocessor';
+
+export default class PreprocessorBucket {
+  public static deserialize(view: StaticDataView, env: Env): PreprocessorBucket {
```

Comment:
Currently, I'm not passing the `env` to serialized engine. Instead, I intended the Engine to re-evaluate everything in deserialization and initialization time. The approach is possible and both `Env` and `PreprocessorBucket` (name of this can be changed) can be serialized.

To match the code style with other deserialization functions, I like this approach.

Reply:
This behavior was changed to call `updateEnv` after deserialization.